src/quo/components/buttons/button/style.cljs

```diff
@@ -55,61 +56,64 @@
-         (when disabled?
-           {:opacity 0.3})))
+  (let [disabled-and-rgb-hex? (and disabled? (colors/rgb-hex? background-color))]
+    (merge {:height             size
+            :align-items        :center
+            :justify-content    :center
+            :flex-direction     (if icon-top :column :row)
+            :padding-horizontal (when-not (or icon-only? icon-left icon-right)
+                                  (case size
+                                    56 (if border-color 10 11)
+                                    40 16
+                                    32 12
+                                    24 7
+                                    16))
+            :padding-left       (when-not (or icon-only? icon-left)
+                                  (case size
+                                    56 nil
+                                    40 16
+                                    32 12
+                                    24 8
+                                    16))
+            :padding-right      (when-not (or icon-only? icon-right)
+                                  (case size
+                                    56 nil
+                                    40 16
+                                    32 12
+                                    24 8
+                                    16))
+            :padding-top        (when-not (or icon-only? icon-left icon-right)
+                                  (case size
+                                    56 0
+                                    40 (if border-color 8 9)
+                                    32 (if border-color 4 5)
+                                    24 0
+                                    (if border-color 8 9)))
+            :padding-bottom     (when-not (or icon-only? icon-left icon-right)
+                                  (case size
+                                    56 0
+                                    40 9
+                                    32 5
+                                    24 0
+                                    9))
+            :overflow           :hidden
+            :background-color   (if disabled-and-rgb-hex?
+                                  (colors/hex->rgba background-color 0.3)
+                                  background-color)
+            :border-radius      (if border-radius
+                                  border-radius
+                                  (case size
+                                    56 12
+                                    40 12
+                                    32 10
+                                    24 8
+                                    12))
+            :border-color       border-color
+            :border-width       (when border-color 1)}
+           (when icon-only?
+             {:width size})
+           (when border-color
+             {:border-color border-color
+              :border-width 1})
+           (when-not disabled-and-rgb-hex?
+             {:opacity 0.3}))))
```

Comment:
Fallback to the older opacity approach if the button is disabled and the value of the background has an opacity.